packages/rocketchat-lib/server/methods/getChannelHistory.js

```diff
@@ -49,8 +49,14 @@ Meteor.methods({
 			records = RocketChat.models.Messages.findVisibleByRoomIdBetweenTimestamps(rid, oldest, latest, options).fetch();
 		}
 
+		const UI_Use_Real_Name = RocketChat.settings.get('UI_Use_Real_Name') === true;
+
 		const messages = _.map(records, (message) => {
 			message.starred = _.findWhere(message.starred, { _id: fromUserId });
+			if (message.u && message.u._id && UI_Use_Real_Name) {
+				const user = RocketChat.models.Users.findOneById(message.u._id);
```

Comment:
Same as my other comment, would it be better to fetch all users in result at once, then map the name on to `message.u.name`?